src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -701,7 +699,7 @@ private Task WriteHexInt32Async(int value)
 
         public Task<HttpResponseMessage> SendWithNtProxyAuthAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            if (_pool.UsingProxy && _pool.ProxyCredentials != null)
+            if (_pool.ProxyUri != null && _pool.ProxyCredentials != null)
```

Comment:
Is there a distinction betweeen `_pool.ProxyUri != null` and `Kind == HttpConnectionKind.Proxy`, or here do they mean the same thing?

Reply:
Yes, there is a distinction here.  And this is what the actual fix for the bug is.  In this case, HttpConnectionKind.Proxy is not sufficient.  That would only send Windows auth creds to a proxy when no tunneling is being used.  This change here makes it so it includes all cases of talking with a proxy with both CONNECT tunneling and no tunneling.

Reply:
Thanks.  In that case, I think the previous name of `UsingProxy` has some value, in that it conveys why we're checking ProxyUri != null.  Consider adding back and then using such a property, e.g. `AnyProxyKindInUse` or some such thing.

Reply:
Ok.  I'll look into doing that.